datastore_mongo_test.go

```diff
@@ -634,6 +645,14 @@ func TestMongoUpsertAttributes(t *testing.T) {
 			t.Errorf("attributes mismatch, have: %v\nwant: %v", dev.Attributes, tc.outAttrs)
 		}
 
+		//check timestamp validity
+		//note that mongo stores time with lower precision- custom comparison
+		assert.Equal(t, createdTs.Unix(), dev.CreatedTs.Unix())
```

Comment:
I used `assert.WithinDuration()` with 1s delta to workaround precision issues. You should be able to replace both asserts with that. Make sure to leave the comment about precision loss.